ext/pdo_pgsql/pgsql_statement.c

```diff
@@ -702,12 +702,28 @@ static int pgsql_stmt_get_col(pdo_stmt_t *stmt, int colno, zval *result, enum pd
 	return 1;
 }
 
-static zend_always_inline char * pdo_pgsql_translate_oid_to_table(Oid oid, PGconn *conn)
+static zend_always_inline char * pdo_pgsql_translate_oid_to_table(Oid oid, pdo_pgsql_db_handle *H)
 {
+	PGconn *conn = H->server;
 	char *table_name = NULL;
 	PGresult *tmp_res;
 	char *querystr = NULL;
 
+	if (oid == H->cached_table_oid) {
+		return H->cached_table_name;
+	}
+
+	if (H->running_stmt) {
+		/* in single-row mode, libpq forbids passing a new query
+		 * while we're still flushing the current one's */
+		return 0;
```

Comment:
nit: `return NULL` would be preferable (yes I know it s like that for the rest of the call).

Reply:
I hesitated, so your advice helped tip the balance.

But by the way I had to recommit, as the CI detected my buggy confusion between "lazy statements" and "all statements": not selective enough I truncated the meta for the "classic" queries too.

… And so I thought of a third problem, cancelling the preceding query if _a new one_ entered the lazy fetching, instead of if _the preceding one_ was lazy and still unfinished.